graphdatascience/procedure_surface/arrow/catalog/relationship_arrow_endpoints.py

```diff
@@ -0,0 +1,191 @@
+
+        job_id = JobClient.run_job(self._arrow_client, endpoint, config)
+        result = JobClient.stream_results(self._arrow_client, G.name(), job_id)
+
+        return result
+
+    def write(
+        self,
+        G: Graph,
+        relationship_type: str,
+        relationship_properties: Optional[list[str]] = None,
+        *,
+        concurrency: Optional[Any] = None,
+        write_concurrency: Optional[Any] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        job_id: Optional[Any] = None,
+    ) -> RelationshipsWriteResult:
+        if self._write_back_client is None:
+            raise ValueError("Write back is only available if a database connection is provided.")
+
+        config_input = {
+            "graph_name": G.name(),
+            "relationship_types": [relationship_type],
+            "concurrency": concurrency,
+            "sudo": sudo,
+            "log_progress": log_progress,
+            "username": username,
+            "job_id": job_id,
+        }
+
+        endpoint = "v2/graph.relationships.stream"
+        if relationship_properties and relationship_properties != []:
+            config_input["relationship_properties"] = relationship_properties
+            endpoint = "v2/graph.relationshipProperties.stream"
+
+        config = ConfigConverter.convert_to_gds_config(**config_input)
+
+        job_id = JobClient.run_job(self._arrow_client, endpoint, config)
+
+        write_result = self._write_back_client.write(
+            G.name(),
+            job_id,
+            concurrency=write_concurrency if write_concurrency is not None else concurrency,
+            relationship_type_overwrite=relationship_type,
+        )
+
+        written_relationships = (
+            write_result.written_relationships if hasattr(write_result, "written_relationships") else 0
+        )
+        written_properties = (
+            written_relationships * len(relationship_properties) if relationship_properties is not None else 0
+        )
+        return RelationshipsWriteResult(
+            graphName=G.name(),
+            relationshipType=relationship_type,
+            relationshipProperties=relationship_properties if relationship_properties is not None else [],
+            relationshipsWritten=written_relationships,
+            propertiesWritten=written_properties,
+            writeMillis=write_result.write_millis,
+            configuration=config,
+        )
+
+    def drop(
+        self,
+        G: Graph,
+        relationship_type: str,
+        *,
+        fail_if_missing: Optional[bool] = None,
+    ) -> RelationshipsDropResult:
+        # TODO enable once we have a propper graph implementation for arrow endpoints
+        # if not relationship_type in G.relationship_types():
+        #     raise ValueError(f"Relationship type '{relationship_type}' does not exist in graph '{G.name()}'")
```

Comment:
we can do this on this PR if we rebase on top of Florentin's, or we do it after both are merged